### PR TITLE
Sqlite3 to sqlite

### PR DESCRIPTION
**Pull request by ChrisCarterCodes**
converting to Promisified library and adding a lot of the auto-vote handling logic

### CODE REVIEW

File: bot-nonpromise.js [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+const config = require('config');
+const sqlite = require('sqlite');
+const dbPromise = sqlite.open('weights', { Promise });
+console.log(typeof dbPromise);
+const db = initialize();
+async function initialize(){
+  try{
+  let db = await dbPromise;
+  console.log(` db: ${db}`);
+  db.run("CREATE TABLE IF NOT EXISTS userVotes (userName TEXT , categoryName TEXT )");
```
Comment (BrandonCookeDev) [inline bot-nonpromise.js]: move inline sql to the sql config file

File: bot-nonpromise.js [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+  //db.run("drop table userinfo");
+  db.run("CREATE TABLE IF NOT EXISTS userInfo (userName TEXT PRIMARY KEY , availableWheelVotes INT DEFAULT 0 )");
```
Comment (BrandonCookeDev) [inline bot-nonpromise.js]: move inline sql to the sql config file

File: bot-nonpromise.js [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+  if (self) { return; } // Ignore messages from the bot
+
+  // Remove whitespace from chat message
+  const commandName = msg.trim();
+  const commandParts= commandName.split(" ");
+
+  // determine permission level
+  var modFlag=isMod(context);
+
+  if(commandParts[0].toLowerCase() == '!bet'  ){
```
Comment (BrandonCookeDev) [inline bot-nonpromise.js]: This if string used to be a switch, what happened

File: bot-nonpromise.js [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+                        "random weapon": {"count": 0, "users": null},
+                        "swordless": {"count": 0, "users": null}
+                      },
+                      "Difficulty":{
+                        "normal": {"count": 0, "users": null},
+                        "hard": {"count": 0, "users": null} 
+                      }
+                    }
+  var message= "Current Votes: "
+  var sql= ("SELECT categoryName, count(categoryName) as counts, group_concat(userName) as users FROM userVotes group by categoryName order by categoryName");
```
Comment (BrandonCookeDev) [inline bot-nonpromise.js]: move inline sql to the sql config file

**Review by BrandonCookeDev — CHANGES_REQUESTED** [review]
It feels like you didn't merge Integration into your current branch and as a result your branches are really out of sync

File: config/default.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (BrandonCookeDev) [inline config/default.json]: what is this? Old config? 
Currently we work off of config.js which consumes environment variables.

File: package.json [hunk 6]
```diff
@@ -4,7 +4,8 @@
   "description": "Bot for handling custom behaviors on Twitch stream",
   "main": "bot.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "set NODE_ENV=prod&& node bot.js"
```
Comment (BrandonCookeDev) [inline package.json]: does this actually work? Does `SET` in bash put the variable into env? Wouldn't you need `export`?

**Review by BrandonCookeDev — CHANGES_REQUESTED** [review]
Few remaining things. PR doesn't seem to include async IIFE wrapped main? Make sure that's in your bot.js

File: config/default.json (hunk 5 above)
Comment (ChrisCarterCodes) [reply config/default.json]: Old config, could be cleaned out

File: package.json (hunk 6 above)
Comment (ChrisCarterCodes) [reply package.json]: Worked for older setup when config was setting twitch settings. really just used in development to npm start the bot. deprecated

**Review by BrandonCookeDev — APPROVED** [review]

**Pull request merged**